src/main/webapp/app/shared/markdown-editor/monaco/markdown-editor-monaco.component.ts

```diff
@@ -0,0 +1,362 @@
+    @ViewChild('resizablePlaceholder', { static: false }) resizePlaceholder?: ElementRef<HTMLDivElement>;
+    @ViewChild('resizeHandle', { static: false }) resizeHandle?: ElementRef<HTMLDivElement>;
+    @ViewChild('actionPalette', { static: false }) actionPalette?: ElementRef<HTMLElement>;
+    @ViewChild(ColorSelectorComponent, { static: false }) colorSelector: ColorSelectorComponent;
+
+    @Input()
+    set markdown(value: string | undefined) {
+        this._markdown = value;
+        this.monacoEditor?.setText(value ?? '');
+    }
+
+    _markdown?: string;
+
+    @Input()
+    enableFileUpload = true;
+
+    @Input()
+    enableResize = true;
+
+    /**
+     * The initial height the editor should have. If set to 'external', the editor will try to grow to the available space.
+     */
+    @Input({ required: true })
+    initialEditorHeight: MarkdownEditorHeight | 'external';
+
+    @Input()
+    resizableMaxHeight = MarkdownEditorHeight.LARGE;
+
+    @Input()
+    resizableMinHeight = MarkdownEditorHeight.SMALL;
+
+    @Input()
+    defaultActions: MonacoEditorAction[] = [
+        new MonacoBoldAction(),
+        new MonacoItalicAction(),
+        new MonacoUnderlineAction(),
+        new MonacoQuoteAction(),
+        new MonacoCodeAction(),
+        new MonacoCodeBlockAction(),
+        new MonacoUrlAction(),
+        new MonacoAttachmentAction(),
+        new MonacoOrderedListAction(),
+        new MonacoUnorderedListAction(),
+    ];
+
+    @Input()
+    headerActions: MonacoEditorActionGroup<MonacoHeadingAction> = new MonacoEditorActionGroup<MonacoHeadingAction>(
+        'artemisApp.multipleChoiceQuestion.editor.style',
+        [1, 2, 3].map((level) => new MonacoHeadingAction(level)),
+        undefined,
+    );
+
+    @Input()
+    colorAction?: MonacoColorAction = new MonacoColorAction();
+
+    @Input()
+    domainActions: MonacoEditorDomainAction[] = [];
+
+    @Input()
+    metaActions: MonacoEditorAction[] = [new MonacoFullscreenAction()];
+
+    @Output()
+    markdownChange = new EventEmitter<string>();
+
+    @Output()
+    onPreviewSelect = new EventEmitter();
+
+    inPreviewMode = false;
+    uniqueMarkdownEditorId: string;
+    faQuestionCircle = faQuestionCircle;
+    faGripLines = faGripLines;
+    resizeObserver?: ResizeObserver;
+    targetWrapperHeight?: number;
+    constrainDragPositionFn?: (pointerPosition: Point) => Point;
+    isResizing = false;
+    displayedActions: MarkdownActionsByGroup = {
+        standard: [],
+        header: [],
+        color: undefined,
+        domain: { withoutOptions: [], withOptions: [] },
+        meta: [],
+    };
+
+    readonly colorToClassMap = new Map<string, string>([
+        ['#ca2024', 'red'],
+        ['#3ea119', 'green'],
+        ['#ffffff', 'white'],
+        ['#000000', 'black'],
+        ['#fffa5c', 'yellow'],
+        ['#0d3cc2', 'blue'],
+        ['#b05db8', 'lila'],
```

Comment:
Is `lila` an English term? 

```suggestion
        ['#b05db8', 'purple'],
```

Reply:
I think you're right that it's not a word. The way the markdown preview works is that we have a few predefined colors:

![image](https://github.com/ls1intum/Artemis/assets/38403547/04a7ea20-d9bb-4408-b343-c18883c16170)

The suffix is then used as the class name in the inserted text (`<span class="lila"></span>`). We could introduce `purple` as a name, but we'd also have to keep `lila` to preserve compatibility with old problem statements. This is why I opted to keep the original

Reply:
I see.